### PR TITLE
fix(splitter): sort dependencies after computing them

### LINKED ISSUE #78: vendor.js is non-deterministic
I'm running into problems with `ember-auto-import` generating a non-deterministic `vendor.js` file.

For reasons beyond my control (😢), our Ember app is built two times on two different instances of the same AWS OpsWorks App Server. Yeah, _I know._

Anyway, both generated `dist` directories are exact carbon copies of one another, except for the `vendor.js`, which seems to have slight differences in module order, leading to different fingerprints in the file name.

For instance in the first line, you can see that the order of `lodash.merge` and `lodash.frompairs` is reversed.

```diff
11:13 $ colordiff vendor-6e637a855fb1298955e4f0ded1e9d3ee.js vendor-71b6d8a2bc05fdef2699448211deaa5e.js
4455c4455
< e.emberAutoImportDynamic=function(e){return r("_eai_dyn_"+e)},t("moment",[],function(){return n(0)}),t("lodash.merge",[],function(){return n(128)}),t("lodash.frompairs",[],function(){return n(126)})}()}});(window.webpackJsonp_ember_auto_import_=window.webpackJsonp_ember_auto_import_||[]).push([[0],[function(e,t,n){(function(e){e.exports=function(){"use strict"
---
> e.emberAutoImportDynamic=function(e){return r("_eai_dyn_"+e)},t("moment",[],function(){return n(0)}),t("lodash.frompairs",[],function(){return n(128)}),t("lodash.merge",[],function(){return n(127)})}()}});(window.webpackJsonp_ember_auto_import_=window.webpackJsonp_ember_auto_import_||[]).push([[0],[function(e,t,n){(function(e){e.exports=function(){"use strict"
5228,5229c5228
< n(0).defineLocale("af",{months:"Januarie_Februarie_Maart_April_Mei_Junie_Julie_Augustus_September_Oktober_November_Desember".split("_"),monthsShort:"Jan_Feb_Mrt_Apr_Mei_Jun_Jul_Aug_Sep_Okt_Nov_Des".split("_"),weekdays:"Sondag_Maandag_Dinsdag_Woensdag_Donderdag_Vrydag_Saterdag".split("_"),weekdaysShort:"Son_Maa_Din_Woe_Don_Vry_Sat".split("_"),weekdaysMin:"So_Ma_Di_Wo_Do_Vr_Sa".split("_"),meridiemParse:/vm|nm/i,isPM:function(e){return/^nm$/i.test(e)},meridiem:function(e,t,n){return e<12?n?"vm":"VM":n?"nm":"NM"},longDateFormat:{LT:"HH:mm",LTS:"HH:mm:ss",L:"DD/MM/YYYY",LL:"D MMMM YYYY",LLL:"D MMMM YYYY HH:mm",LLLL:"dddd, D MMMM YYYY HH:mm"},calendar:{sameDay:"[Vandag om] LT",nextDay:"[Môre om] LT",nextWeek:"dddd [om] LT",lastDay:"[Gister om] LT",lastWeek:"[Laas] dddd [om] LT",sameElse:"L"},relativeTime:{future:"oor %s",past:"%s gelede",s:"'n paar sekondes",ss:"%d sekondes",m:"'n minuut",mm:"%d minute",h:"'n uur",hh:"%d ure",d:"'n dag",dd:"%d dae",M:"'n maand",MM:"%d maande",y:"'n jaar",yy:"%d jaar"},dayOfMonthOrdinalParse:/\d{1,2}(ste|de)/,ordinal:function(e){return e+(1===e||8===e||e>=20?"ste":"de")},week:{dow:1,doy:4}})}()},function(e,t){e.exports=function(e){return e.webpackPolyfill||(e.deprecate=function(){},e.paths=[],e.children||(e.children=[]),Object.defineProperty(e,"loaded",{enumerable:!0,get:function(){return e.l}}),Object.defineProperty(e,"id",{enumerable:!0,get:function(){return e.i}}),e.webpackPolyfill=1),e}},,function(e,t){e.exports=function(e){for(var t=-1,n=e?e.length:0,r={};++t<n;){var i=e[t]
< r[i[0]]=i[1]}return r}},function(e,t){var n
---
> n(0).defineLocale("af",{months:"Januarie_Februarie_Maart_April_Mei_Junie_Julie_Augustus_September_Oktober_November_Desember".split("_"),monthsShort:"Jan_Feb_Mrt_Apr_Mei_Jun_Jul_Aug_Sep_Okt_Nov_Des".split("_"),weekdays:"Sondag_Maandag_Dinsdag_Woensdag_Donderdag_Vrydag_Saterdag".split("_"),weekdaysShort:"Son_Maa_Din_Woe_Don_Vry_Sat".split("_"),weekdaysMin:"So_Ma_Di_Wo_Do_Vr_Sa".split("_"),meridiemParse:/vm|nm/i,isPM:function(e){return/^nm$/i.test(e)},meridiem:function(e,t,n){return e<12?n?"vm":"VM":n?"nm":"NM"},longDateFormat:{LT:"HH:mm",LTS:"HH:mm:ss",L:"DD/MM/YYYY",LL:"D MMMM YYYY",LLL:"D MMMM YYYY HH:mm",LLLL:"dddd, D MMMM YYYY HH:mm"},calendar:{sameDay:"[Vandag om] LT",nextDay:"[Môre om] LT",nextWeek:"dddd [om] LT",lastDay:"[Gister om] LT",lastWeek:"[Laas] dddd [om] LT",sameElse:"L"},relativeTime:{future:"oor %s",past:"%s gelede",s:"'n paar sekondes",ss:"%d sekondes",m:"'n minuut",mm:"%d minute",h:"'n uur",hh:"%d ure",d:"'n dag",dd:"%d dae",M:"'n maand",MM:"%d maande",y:"'n jaar",yy:"%d jaar"},dayOfMonthOrdinalParse:/\d{1,2}(ste|de)/,ordinal:function(e){return e+(1===e||8===e||e>=20?"ste":"de")},week:{dow:1,doy:4}})}()},function(e,t){e.exports=function(e){return e.webpackPolyfill||(e.deprecate=function(){},e.paths=[],e.children||(e.children=[]),Object.defineProperty(e,"loaded",{enumerable:!0,get:function(){return e.l}}),Object.defineProperty(e,"id",{enumerable:!0,get:function(){return e.i}}),e.webpackPolyfill=1),e}},,function(e,t){var n
5321c5320,5321
< function xe(e){return e}n.exports=Ee}).call(this,n(127),n(124)(e))}]]),"undefined"==typeof FastBoot&&function(e,t){"object"==typeof exports&&"object"==typeof module?module.exports=t():"function"==typeof define&&define.amd?define([],t):"object"==typeof exports?exports.ClipboardJS=t():e.ClipboardJS=t()}(this,function(){return function(e){var t={}
---
> function xe(e){return e}n.exports=Ee}).call(this,n(126),n(124)(e))},function(e,t){e.exports=function(e){for(var t=-1,n=e?e.length:0,r={};++t<n;){var i=e[t]
> r[i[0]]=i[1]}return r}}]]),"undefined"==typeof FastBoot&&function(e,t){"object"==typeof exports&&"object"==typeof module?module.exports=t():"function"==typeof define&&define.amd?define([],t):"object"==typeof exports?exports.ClipboardJS=t():e.ClipboardJS=t()}(this,function(){return function(e){var t={}
8297c8297
< //# sourceMappingURL=/assets/client/assets/vendor-2d8eaa94d883db59750746c87931b646.map
\ No newline at end of file
---
> //# sourceMappingURL=/assets/client/assets/vendor-84ffd757cb1627f52afc797f9e08aac0.map
\ No newline at end of file
```

This has been working fine for us for quite some time and just recently started breaking. Unfortunately, I can't exactly pin down when.

We are on `ember-auto-import@1.2.3` and use yarn, so dependencies are guaranteed to be the same on both instances.

Do you have any pointers, where I can start digging?

### PR DESCRIPTION
**Pull request by buschtoens**
Fixes #78.

I've tested this on our prod servers and it seems to work reliably.

@ef4 Thank you so much for the pointers in #78. You massively reduced the time I needed to fix this. ❤️

### CODE REVIEW

**Comment by buschtoens** [comment]
Travis seems to have had connectivity issues. Maybe you can retrigger?

File: ts/splitter.ts [hunk 1]
```diff
@@ -155,9 +155,23 @@ export default class Splitter {
...
 
+  private sortDependencies(deps: Map<string, BundleDependencies>) {
+    for (const bundle of deps.values()) {
+      this.sortBundle(bundle);
+    }
+  }
+
+  private sortBundle(bundle: BundleDependencies) {
+    for (const imports of values(bundle)) {
+      imports.sort((a, b) => a.specifier.localeCompare(b.specifier));
```
Comment (buschtoens) [inline ts/splitter.ts]: As https://github.com/nodejs/node/issues/4067 shows `localeCompare` has already been available pre `6.x`, so we're fine using that here. 😊

**Comment by ef4** [comment]
Thanks, looks good.

**Comment by buschtoens** [comment]
Did you somehow merge this PR without merging it? 😅 

Are you a wizard? 🧙🏼‍♂️

**Comment by ef4** [comment]
GitHub is having a consistency problem. 😝

**Comment by ef4** [comment]
This was merged.

**Pull request closed without merging**